File: kotlin-multiplatform/monstradore/androidApp/src/main/java/com/example/monstradore/android/persistence/PersistenceView.kt
package com.example.monstradore.android.persistence

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.material.TextField
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.monstradore.storage.*

@Composable
fun PersistenceContent(
    storage: UserStorage
) {
    var text by remember { mutableStateOf("") }
    var users by remember { mutableStateOf(storage.getUsers().toSet())}
    Column(
        modifier = Modifier.padding(10.dp)
    ) {
        Text(
            text = "Persistenz",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold
        )
        Text("Username eingeben")
        TextField(
            value = text,
            onValueChange = { text = it },
            label = { Text("Name") }
        )
        Row {
            Button(
                modifier = Modifier.padding(0.dp, 0.dp, 5.dp, 0.dp),
                onClick = {
                    users += text
                    storage.saveUsers(users.toList())
                }
            ) {
                Text("User speichern")
            }
            Button(
                onClick = { }
            ) {
                Text("User laden")
            }
        }
        users.forEach { user ->
            Text(user + "\n")
        }
    }
}

File: kotlin-multiplatform/monstradore/iosApp/iosApp/persistence/PersistenceView.swift
//
//  PersistenceView.swift
//  iosApp
//
//  Created by Anja on 25.11.22.
//  Copyright © 2022 orgName. All rights reserved.
//

import SwiftUI
import shared

struct PersistenceView : View {
    @State private var userName: String = ""
    @State private var users: [String] = UserStorage(storage: NSObject()).getUsers()
    private var storage: UserStorage = UserStorage(storage: NSObject())
    var body: some View {
        VStack{
            Text("Persistenz")
                .font(.headline)
            Text("Username eingeben")
            TextField("Name", text: $userName)
            Button(action: {
                users.append(userName)
                storage.saveUsers(users: users)
            }) {
                Text("User speichern")
            }
            ForEach(users) { user in
                Text(user)
            }
        }
        .padding()
        .frame(alignment: .topLeading)
    }
}

//struct PersistenceView_Previews: PreviewProvider {
//    static var previews: some View {
//        PersistenceView()
//    }
//}
